kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/test/java/org/kie/server/services/jbpm/ui/ImageServiceBaseTest.java

```diff
@@ -191,4 +200,50 @@ public void testLoopSubProcess() throws Exception {
         assertNotNull(style);
         assertEquals("cursor: pointer;", style);
     }
+
+    @Test
+    public void testSignalEventSubProcess() throws Exception {
+        String containerId = "test-container";
+        String processId = "test-processId";
+        long processInstanceId = 10;
+
+        byte[] byteArray = getInputStreamAsByteArray(ImageServiceBaseTest.class.getResourceAsStream("/signal-event-subprocess-svg.svg"));
+
+        when(dataService.getProcessesByDeploymentIdProcessId(containerId, processId)).thenReturn(mock(ProcessDefinition.class));
+        when(imageReference.getImageContent(anyString(), anyString())).thenReturn(byteArray);
+        when(kieServerRegistry.getConfig()).thenReturn(config);
+        when(config.getConfigItemValue(anyString(), anyString())).thenReturn("");
+        final String abortNodeId = "_66CB2BD8-424B-4057-B672-F17B6D6F2AB4";
+        final String completedNodeId = "_9ACAA96A-93EC-4D51-8CF0-EAB1F28F5F56";
+        NodeInstanceDesc nodeInstanceDescCompleted1 = new org.jbpm.kie.services.impl.model.NodeInstanceDesc("1", completedNodeId, "", "StartNode", "", 1L, null,
+                                                                                                            "", 1, 1L, 3L, "", null, 1);
+
+        NodeInstanceDesc nodeInstanceDescAbort = new org.jbpm.kie.services.impl.model.NodeInstanceDesc("2", abortNodeId, "", "SubProcessNode", "", 1L, null,
+                                                                                                       "", 2, 1L, 3L, "", null, 1);
+
+        when(dataService.getProcessInstanceHistoryCompleted(anyLong(), any())).thenReturn(Arrays.asList(nodeInstanceDescCompleted1));
+        when(dataService.getProcessInstanceHistoryActive(anyLong(), any())).thenReturn(Collections.EMPTY_LIST);
+        when(dataService.getProcessInstanceFullHistory(anyLong(), any())).thenReturn(Arrays.asList(nodeInstanceDescAbort));
```

Comment:
@bxf12315 I guess for the testing purposes. getProcessInstanceFullHistory have to return nodes of many types just to test the filter.. FullHistory = Active + Completed + Others. What we need to test is that the 'Others' Poputale subprocess link too.

Reply:
updated